e2e_tests/tests/experiment/test_custom_searcher.py

```diff
@@ -0,0 +1,580 @@
+import logging
+import subprocess
+import tempfile
+import time
+from pathlib import Path
+from typing import List, Optional
+
+import pytest
+import yaml
+from urllib3.connectionpool import HTTPConnectionPool, MaxRetryError
+
+from determined import searcher
+from determined.common.api import bindings
+from determined.common.api.bindings import determinedexperimentv1State
+from determined.experimental import client
+from tests import config as conf
+from tests import experiment as exp
+from tests.fixtures.custom_searcher import searchers
+
+TIMESTAMP = int(time.time())
+
+
+@pytest.mark.e2e_cpu
+def test_run_custom_searcher_experiment(tmp_path: Path) -> None:
+    # example searcher script
+    config = conf.load_config(conf.fixtures_path("no_op/single.yaml"))
+    config["searcher"] = {
+        "name": "custom",
+        "metric": "validation_error",
+        "smaller_is_better": True,
+        "unit": "batches",
+    }
+    config["name"] = "single"
+    config["description"] = "custom searcher"
+    search_method = searchers.SingleSearchMethod(config, 500)
+    search_runner = searcher.LocalSearchRunner(search_method, tmp_path)
+    experiment_id = search_runner.run(config, model_dir=conf.fixtures_path("no_op"))
+
+    assert client._determined is not None
+    session = client._determined._session
+    response = bindings.get_GetExperiment(session, experimentId=experiment_id)
+    assert response.experiment.numTrials == 1
+
+
+@pytest.mark.e2e_cpu_2a
+def test_run_random_searcher_exp() -> None:
+    config = conf.load_config(conf.fixtures_path("no_op/single.yaml"))
+    config["searcher"] = {
+        "name": "custom",
+        "metric": "validation_error",
+        "smaller_is_better": True,
+        "unit": "batches",
+    }
+    config["name"] = "random"
+    config["description"] = "custom searcher"
+
+    max_trials = 5
+    max_concurrent_trials = 2
+    max_length = 500
+
+    with tempfile.TemporaryDirectory() as searcher_dir:
+        search_method = searchers.RandomSearchMethod(
+            max_trials, max_concurrent_trials, max_length, test_type="noop"
+        )
+        search_runner = searcher.LocalSearchRunner(search_method, Path(searcher_dir))
+        experiment_id = search_runner.run(config, model_dir=conf.fixtures_path("no_op"))
+
+    assert client._determined is not None
+    session = client._determined._session
+    response = bindings.get_GetExperiment(session, experimentId=experiment_id)
+    assert response.experiment.numTrials == 5
+    assert search_method.created_trials == 5
+    assert search_method.pending_trials == 0
+    assert search_method.closed_trials == 5
+    assert len(search_runner.state.trials_created) == search_method.created_trials
+    assert len(search_runner.state.trials_closed) == search_method.closed_trials
+
+
+@pytest.mark.e2e_cpu_2a
+@pytest.mark.parametrize(
```

Comment:
question: how long does this set of e2e tests take to run?

I like the failure testing, but I'm a bit concerned about e2e test costs (developer time more than cloud costs)